TradeItIosTicketSDK2/TradeItTransactionPresenter.swift

```diff
@@ -41,7 +41,14 @@ class TradeItTransactionPresenter: NSObject {
         
         return "\(formatLabel(action)) \(quantity) \(symbol) @ \(price)"
     }
-    
+
+    func getTransactionDescriptionLabel() -> String {
+        guard self.transaction.typeEnum != .trade else {
+            return getDescriptionLabel()
+        }
+        return self.transaction.transactionDescription ?? ""
```

Comment:
We should probably return a `String?` here so you don't use the defaults all the time. The `null` case has meaning in that you won't be adding the row to be displayed. Otherwise you need to do all those `!= ""` checks above which is strange.

Reply:
We don't want to display if it is empty so I will have to check this in the if like:
`let transactionLabel = if self.presenter?.getTransactionActionLabel() && transactionLabel != ""` ?
Or do you have another idea ?